.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, master-2.x ]
```

Comment:
Do we really need it on push?

Reply:
I think it needs some reference point to compare new PRs with. Frankly, this file is autogenerated by GitHub, I just wanted to see whether it finds anything and it actually did find some (very minor, but still) things to improve.